beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -1840,13 +1840,16 @@ impl<E: EthSpec> Network<E> {
                     }
                 }
                 SwarmEvent::ListenerError { error, .. } => {
-                    // this is non fatal, but we still check
-                    warn!(self.log, "Listener error"; "error" => ?error);
-                    if Swarm::listeners(&self.swarm).count() == 0 {
-                        Some(NetworkEvent::ZeroListeners)
-                    } else {
-                        None
+                    // Ignore quic accept and close errors.
+                    if error
+                        .get_ref()
+                        .and_then(|err| err.downcast_ref::<libp2p::quic::Error>())
+                        .filter(|err| matches!(err, libp2p::quic::Error::Connection(_)))
+                        .is_none()
+                    {
+                        warn!(self.log, "Listener error"; "error" => ?error);
                     }
+                    None
```

Comment:
previously we were checking the number of listeners, but it's redundant here as they will only decrease if gracefully closed, `ListenerError`s are non fatal and therefore shouldn't close the listener